stellargraph/mapper/graphwave_generator.py

```diff
@@ -64,14 +64,14 @@ def __init__(self, G, scales="auto", num_scales=3, num_eigenvecs=-1):
         # Function to map node IDs to indices for quicker node index lookups
         self._node_lookup = G._get_index_for_nodes
 
-        degree_mat = diags(np.array(adj.sum(1)).flatten())
+        degree_mat = diags(np.asarray(adj.sum(1)).ravel())
         laplacian = degree_mat - adj
 
         if num_eigenvecs == -1:
             num_eigenvecs = laplacian.shape[0] - 2
 
         # TODO: add in option to compute wavelet transform using Chebysev polynomials
-        eigen_vals, eigen_vecs = eigs(laplacian, k=num_eigenvecs)
+        eigen_vals, eigen_vecs = eigs(laplacian, k=num_eigenvecs, which="SM")
```

Comment:
Maybe we could have an explanatory comment about why `"SM"` here?

From testing this in the Cora notebook in my other comment:

- this seems to be very slow and very variable (e.g. k=32 has taken from 10-34 seconds across few runs with `which="SM"` vs. fairly reliably ~30 milliseconds with `which="LM"`)
- it seems we may still have to compute quite a few of the eigenvalues to get a reasonable norm of `exp(-s eigenvalues)`? E.g. with `which="SM"` the norm table up to `k=64` of the Cora notebook looks like:

  ![image](https://user-images.githubusercontent.com/1203825/74393288-c909a900-4e5d-11ea-8df6-b4c666df50d9.png)

  That is, it's still 7.80 vs. 10.2 for `k=64`.

Reply:
- the thinking was that even though it would be slower, having to calculate less eigenvalues would make the whole algorithm faster overall. There's some performance enhancing options I'll explore